mirgecom/profiling.py

```diff
@@ -253,7 +278,7 @@ def tabulate_profiling_data(self) -> pytools.Table:
                 bandwidth_access_mean = "--"
                 bandwidth_access_max = "--"
```

Comment:
This code could probably just use `get_profiling_data_for_kernel` now, to avoid code duplication.

Reply:
I'm not sure this would be too useful. `get_profiling_data_for_kernel` only returns averages, while `tabulate_profiling_data` also calculates min/max as well as derived statistics such as bandwidth etc. Imho we wouldn't gain much by using `get_profiling_data_for_kernel` here.

Reply:
I mean, that's kind of a strange inconsistency, too. Why can one gather max/min, the other not?

Reply:
We can modify `get_profiling_data_for_kernel` to also log min/max, if you prefer.

Reply:
How about returning something like this:
```python
class StatisticsAccumulator:
    def __init__(self):
        self.num_values = 0
        self._sum = 0
        self._min = 0
        self._max = 0

    def add_value(self, v):
        if v is None:
             return
        self.num_values += 1
        self._sum += v
        # ...

    def mean(self):
        if self.num_values == 0:
            return None

        return self.sum / self.num_values
# ...
```

Reply:
Just as a side note, I think my original idea for just doing averages here was to reduce the sheer number of statistics and the associated clutter (+overhead). Just adding min/max would triple the amount of kernel profiling data, which is already by far the majority of data stored for each time step. Also, it would create nice-looking entries like `multiply_time_min.max`.

Reply:
Sure! I'm not saying we have to create `LogQuantity`s for all of them. This accumulator thing is just so that we can unify the summation logic.

Reply:
Added the solution you outlined. This works a bit differently for the derived quantities